db/case-sql.ts

```diff
@@ -0,0 +1,155 @@
+        END
+      AND
+        CASE WHEN $<phoneNumber> IS NULL THEN TRUE
+        ELSE (
+          ${phoneNumberColumns.map(pn => `regexp_replace(${pn}, '\\D', '', 'g') ILIKE $<phoneNumber>`).join('\n OR ')}
+        )
+        END`
+
+const SEARCH_WHERE_CLAUSE = `WHERE
+    (info IS NULL OR jsonb_typeof(info) = 'object')
+    AND
+      CASE WHEN $<helpline> IS NULL THEN TRUE
+      ELSE  (
+        cases.helpline = $<helpline>
+      )
+      END
+    AND
+      $<accountSid> IS NOT NULL AND cases."accountSid" = $<accountSid>
+    AND
+      CASE WHEN $<counselor> IS NULL THEN TRUE
+      ELSE cases."twilioWorkerId" = $<counselor>
+      END
+    AND
+      CASE WHEN $<dateFrom> IS NULL THEN TRUE
+      ELSE cases."createdAt"::DATE >= $<dateFrom>::DATE
+      END
+    AND
+      CASE WHEN $<dateTo> IS NULL THEN TRUE
+      ELSE cases."createdAt"::DATE <= $<dateTo>::DATE
+      END
+    AND (
+      -- search on childInformation of connectedContacts
+      (
+        ${nameAndPhoneNumberSearchSql("contacts.\"rawJson\"->'childInformation'->'name'->>'firstName'", "contacts.\"rawJson\"->'childInformation'->'name'->>'lastName'", ["contacts.\"rawJson\"->'childInformation'->'location'->>'phone1'", "contacts.\"rawJson\"->'childInformation'->'location'->>'phone2'", "contacts.number"])}  
+      )
+        -- search on callerInformation of connectedContacts
+      OR ( 
+        ${nameAndPhoneNumberSearchSql("contacts.\"rawJson\"->'callerInformation'->'name'->>'firstName'", "contacts.\"rawJson\"->'callerInformation'->'name'->>'lastName'", ["contacts.\"rawJson\"->'callerInformation'->'location'->>'phone1'", "contacts.\"rawJson\"->'callerInformation'->'location'->>'phone2'", "contacts.number"])}  
+      )
+        -- search on households
+      OR (
+        ${nameAndPhoneNumberSearchSql("h.value->'household'->>'firstName'", "h.value->'household'->>'lastName'", ["h.value->'household'->>'phone1'", "h.value->'household'->>'phone2'"])}
+       )
+  
+        -- search on perpetrators
+      OR (
+        ${nameAndPhoneNumberSearchSql("p.value->'perpetrator'->>'firstName'", "p.value->'perpetrator'->>'lastName'", ["p.value->'perpetrator'->>'phone1'", "p.value->'perpetrator'->>'phone2'"])}
+      )
+    )
+    -- previous contacts search
+    AND
+      CASE WHEN $<contactNumber> IS NULL THEN TRUE
+      ELSE contacts.number = $<contactNumber>
+      END`
+
+const SEARCH_HAVING_CLAUSE = `  
+  -- Needed a HAVING clause because we couldn't do aggregations on WHERE clauses
+  HAVING
+    -- search on closedCases and orphaned cases (without connected contacts)
+    CASE WHEN $<closedCases>::BOOLEAN = FALSE THEN (
+      cases.status <> 'closed'
+      AND jsonb_array_length(COALESCE(jsonb_agg(DISTINCT contacts.*) FILTER (WHERE contacts.id IS NOT NULL), '[]')) > 0
+    )
+    ELSE TRUE
+    END
+`
+
+const selectSingleCaseByIdSql = (tableName: string) => `SELECT
+        cases.*,
+        contacts."connectedContacts"
+        FROM "${tableName}" AS cases
```

Comment:
It is also used in the update query, when returning the updated value - but it references the special 'updated' table returned as part of the update operatrion, rather than the 'Cases' table directly